File: spring-backend/src/main/java/com/scanmyfood/backend/models/ImageGenerationResponse.java
package com.scanmyfood.backend.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ImageGenerationResponse {
    private String imageData; // Base64-encoded image
    private String mimeType;   // e.g., "image/png"
    private String description; // Original food description
    private int sizeBytes;      // Image size in bytes
}

File: flutter-app/lib/models/image_generation_response.dart
class ImageGenerationResponse {
  final String imageData; // Base64-encoded image
  final String mimeType;
  final String description;
  final int sizeBytes;

  ImageGenerationResponse({
    required this.imageData,
    required this.mimeType,
    required this.description,
    required this.sizeBytes,
  });

  factory ImageGenerationResponse.fromJson(Map<String, dynamic> json) {
    return ImageGenerationResponse(
      imageData: json['image_data'] as String,
      mimeType: json['mime_type'] as String,
      description: json['description'] as String,
      sizeBytes: json['size_bytes'] as int,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'image_data': imageData,
      'mime_type': mimeType,
      'description': description,
      'size_bytes': sizeBytes,
    };
  }
}
